### PR TITLE
Add error detail if query contains both local and distributed tables

### PR DESCRIPTION
**Pull request by velioglu**
Add error detail if query has both local and distributed tables, since Citus supports `INSERT INTO dist_table SELECT ... FROM local_table` queries right now.

### CODE REVIEW

File: src/backend/distributed/planner/multi_logical_planner.c [hunk 1]
```diff
@@ -2864,7 +2864,10 @@ NeedsDistributedPlanning(Query *queryTree)
 	if (hasLocalRelation && hasDistributedRelation)
 	{
 		ereport(ERROR, (errmsg("cannot plan queries which include both local and "
-							   "distributed relations")));
+							   "distributed relations"),
+						errdetail(
+							"Only exception is INSERT INTO dist_table SELECT ... FROM "
```
Comment (marcocitus) [inline src/backend/distributed/planner/multi_logical_planner.c]: Hmz, is there a way to only display a relevant error when the query is specifically `INSERT INTO local_table SELECT ... FROM distributed_table ...`?

Comment (velioglu) [inline src/backend/distributed/planner/multi_logical_planner.c]: By returning the reason of false result from `InsertSelectInfoDistributedTable` function , yes.

Comment (marcocitus) [inline src/backend/distributed/planner/multi_logical_planner.c]: Either that or create a new function.

I think always showing this `errdetail` in all cases might only add to confusion.

File: src/backend/distributed/planner/multi_logical_planner.c [hunk 2]
```diff
@@ -2863,6 +2863,11 @@ NeedsDistributedPlanning(Query *queryTree)
 
 	if (hasLocalRelation && hasDistributedRelation)
 	{
+		if (InsertSelectIntoLocalTable(queryTree))
+		{
+			ereport(ERROR, (errmsg("cannot plan query, Citus only supports INSERT INTO "
+								   "dist_table SELECT ... FROM local_table")));
```
Comment (marcocitus) [inline src/backend/distributed/planner/multi_logical_planner.c]: This is not quite accurate, we also support all sorts of other SELECTs (e.g. `FROM dist_table`). Maybe just say that `INSERT INTO local_table SELECT ... FROM dist_table...` is not supported?

File: src/test/regress/expected/multi_insert_select_non_pushable_queries.out [hunk 3]
```diff
@@ -1,5 +1,15 @@
 ------------------------------------
 ------------------------------------
+-- Insert into local table
+------------------------------------
+------------------------------------
```
Comment (marcocitus) [inline src/test/regress/expected/multi_insert_select_non_pushable_queries.out]: This commenting style is a little unusual.

File: src/backend/distributed/planner/insert_select_planner.c [hunk 4]
```diff
@@ -80,11 +81,51 @@ static void CastSelectTargetList(List *selectTargetList, Oid targetRelationId,
...
+			return true;
+		}
+	}
+
+	return false;
+}
+
+
+bool
+InsertSelectIntoLocalTable(Query *query)
```
Comment (marcocitus) [inline src/backend/distributed/planner/insert_select_planner.c]: These need function comments.

**Comment by metdos** [comment]
@velioglu, is this PR ready for another round of review?

**Comment by velioglu** [comment]
@metdos No, not yet. I'm planning to update the PR today.

File: src/test/regress/expected/multi_insert_select_non_pushable_queries.out (hunk 3 above)
Comment (velioglu) [reply src/test/regress/expected/multi_insert_select_non_pushable_queries.out]: I used that commenting style to be consistent with the other comments in this file.

**Comment by velioglu** [comment]
@marcocitus It is ready for the review. #1632 will solve the errors on Travis.

**Comment by jasonmp85** [comment]
The Travis build for PostgreSQL 11 doesn't affect the health check here. The problem was the 9 and 10 builds were failing because new versions were released so the `initdb` version didn't match the `postgres` one. I had assumed the `enable-depends` flag would have fixed that, but I guess not? Clearing the cache and re-building didn't seem to fix anything…

File: src/backend/distributed/planner/multi_logical_planner.c [hunk 5]
```diff
@@ -2863,6 +2863,11 @@ NeedsDistributedPlanning(Query *queryTree)
 
 	if (hasLocalRelation && hasDistributedRelation)
 	{
+		if (InsertSelectIntoLocalTable(queryTree))
+		{
+			ereport(ERROR, (errmsg("cannot plan INSERT INTO local_table SELECT ... FROM"
```
Comment (jasonmp85) [inline src/backend/distributed/planner/multi_logical_planner.c]: I think this should be reworded as _cannot `INSERT` rows from a distributed query into a local table_. The `dist_table` and `local_table` identifiers are ad hoc and will be misleading appearing in SQL like that.

**Comment by jasonmp85** [comment]
After re-triggering the build it appears there is an actual test failure here.

**Comment by metdos** [comment]
@velioglu, could you rebase and merge this?

File: src/test/regress/expected/multi_colocation_utils.out [hunk 6]
```diff
@@ -266,11 +266,11 @@ SELECT shards_colocated(1300000, 1300020);
 (1 row)
 
 -- check co-located table list
-SELECT UNNEST(get_colocated_table_array('table1_group1'))::regclass;
+SELECT UNNEST(get_colocated_table_array('table1_group1'))::regclass ORDER BY 1;
```
Comment (jasonmp85) [inline src/test/regress/expected/multi_colocation_utils.out]: Uuuuggghhh 👎… we now have this change in three separate branches. How did this get checked in in the first place?!?

Lemme merge #1667 and you can rebase on that and remove this commit.

Comment (velioglu) [reply src/test/regress/expected/multi_colocation_utils.out]: Got the shipit before your PR I guess. Waiting for your merge.

**Pull request merged**